games/hangman.c

```diff
@@ -0,0 +1,277 @@
+        if (tolower(word[i]) == letter) {
+            return 1;
+        }
+    }
+
+    return -1;
+}
+
+/**
+ * @brief creates a new game - generates a random word and stores in global variable current_word
+ * @returns void
+ */
+void new_game() {
+
+    int line_number = 0;
+    int random_num;
+    int s = 0;
+
+    FILE *fptr;
+    fptr = fopen("words.txt", "r");
+
+    // counts number of words in file - assumes each word on new line
+    while (fgets(current_word, 100, fptr) != NULL) {
+        line_number++;
+    }
+    fclose(fptr);
+
+    srand(time(NULL));
+    random_num = rand() % line_number;
+
+    // selects randomly generated word
+    fptr = fopen("words.txt", "r");
+    while (s < random_num){
+        fgets(current_word, 100, fptr);
+        s++;
+    }
+    fclose(fptr);
+}
+
+/**
+ * @brief checks if player has won or lost
+ * @param score how many incorrect guesses player has made
+ * @returns void
+ */
+void won(int score) {
+    if (score > 12) {
+        printf("\nYou lost! The word was: %s", current_word);
+    }
+    else {
+        printf("\nYou won! You had %d guesses left", (13 - score));
+    }
+}
+
+/*
+ * @brief gradually draws duck as user gets letters incorrect
+ * @param score how many incorrect guesses player has made
+ * @returns void
+ */
+void picture(int score) {
+
+    switch(score) {
+
+        case 11:
+            printf("\n      _\n"
+                   "  __( ' )\n"
+                   " \\_ < _ ) ");
+            break;
+
+        case 10:
+            printf("\n      _\n"
+                   "  __(   )\n"
+                   " \\_ < _ ) ");
+            break;
+
+        case 9:
+            printf("\n        \n"
+                   "  __(   )\n"
+                   " \\_ < _ ) ");
+            break;
+
+        case 8:
+            printf("\n        \n"
+                   "  __(    \n"
+                   " \\_ < _ ) ");
+            break;
+
+        case 7:
+            printf("\n        \n"
+                   "  __     \n"
+                   " \\_ < _ ) ");
+            break;
+
+        case 6:
+            printf("\n        \n"
+                   "  _      \n"
+                   " \\_ < _ ) ");
+            break;
+
+        case 5:
+            printf("\n        \n"
+                   "  _      \n"
+                   "   _ < _ ) ");
+            break;
+
+        case 4:
+            printf("\n        \n"
+                   "         \n"
+                   "   _ < _ ) ");
+            break;
+
+        case 3:
+            printf("\n        \n"
+                   "         \n"
+                   "     < _ ) ");
+            break;
+
+        case 2:
+            printf("\n        \n"
+                   "         \n"
+                   "       _ ) ");
+            break;
+
+        case 1:
+            printf("\n        \n"
+                   "         \n"
+                   "         ) ");
+            break;
+
+        case 0:
+            break;
+
+        default:
```

Comment:
This would activate when the player run out of guesses AND when they have only one guess left, not sure if that's intended.